enhancements/olm-deletes-operands.md

```diff
@@ -0,0 +1,209 @@
+---
+title: olm-deletes-operands
+authors:
+  - "@hasbro17"
+reviewers:
+  - "@benluddy"
+  - "@ecordell"
+  - "@kevinrizza"
+  - "@njhale"
+approvers:
+  - "@kevinrizza"
+  - "@dmesser"
+  - "@ecordell"
+  - "@shawn-hurley"  
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional
+see-also:
+  - "N/A"
+replaces:
+  - "N/A"
+superseded-by:
+  - "N/A"
+---
+
+# olm-deletes-operands
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The purpose of this enhancement is to outline how the [Operator Lifecycle Manager (OLM)][olm] can let operator users opt-in to the cleanup of their operator workloads, which would be cleaned up upon the uninstallation of their operator. Specifically when uninstalling an operator, OLM would delete all CustomResource (CR) objects managed by an operator so that the operator can be triggered to clean up all the resources associated with its CRs via [Finalizers][finalizers].
```

Comment:
Since the goal is to trigger cleanup of resources via finalizers, deleting the CRD is not strictly necessary.
And deleting the CRD is not always possible when you have namespace scoped operators. I have blurb on this in a later section: [CRD removal and caveats](https://github.com/operator-framework/enhancements/pull/46/files#diff-2c4dde2fa9d8e866e30702506f1b86bbef00c2c39f6ea144b0c6df3dd6635166R170)

> If there are two non-global operators installed of the same type (providing the same APIs), then removing the CRD during the cleanup of one operator would result in the deletion of CRs for the other operator as well.

My intent was to keep the summary brief without going into the reasons but not sure if you want me to throw a link to that section here, or if that section needs more details.